ddtrace/contrib/internal/ray/span_manager.py

```diff
@@ -0,0 +1,241 @@
+import atexit
+from contextlib import contextmanager
+from itertools import chain
+import threading
+import time
+
+from ddtrace import config
+from ddtrace import tracer
+from ddtrace._trace.span import Span
+from ddtrace.constants import SPAN_KIND
+from ddtrace.ext import SpanKind
+
+from .utils import _inject_ray_span_tags_and_metrics
+
+
+@contextmanager
+def long_running_ray_span(span_name, service, span_type, child_of=None, activate=True):
+    """Context manager that handles Ray span creation and long-running span lifecycle"""
+    with tracer.start_span(
+        name=span_name, service=service, span_type=span_type, child_of=child_of, activate=activate
+    ) as span:
+        span.set_tag_str(SPAN_KIND, SpanKind.CONSUMER)
+        _inject_ray_span_tags_and_metrics(span)
+        start_long_running_span(span)
+
+        try:
+            yield span
+        except BaseException as e:
+            span.set_exc_info(type(e), e, e.__traceback__)
```

Comment:
This should be handled by the span context manager

Reply:
addressed in [936e2f8](https://github.com/DataDog/dd-trace-py/pull/14511/commits/936e2f8879995831232f579c7b92f200c53d0e01)